### ISSUE TITLE
cannot find -ludev: No such file or directory

### ISSUE THREAD

**Issue by katie-hughes** [ngmor/unitree_camera#2]
When I use colcon to build this project, I receive the following errors:

```
Starting >>> UnitreeCameraSDK
Starting >>> unitree_camera_launch_module
Finished <<< unitree_camera_launch_module [1.05s]                    
--- stderr: UnitreeCameraSDK                              
/usr/bin/ld: cannot find -ludev: No such file or directory
collect2: error: ld returned 1 exit status
gmake[2]: *** [examples/CMakeFiles/example_getPointCloud.dir/build.make:192: /home/khughes1/new_unitree_cam/src/UnitreecameraSDK/bins/example_getPointCloud] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:294: examples/CMakeFiles/example_getPointCloud.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
/usr/bin/ld: cannot find -ludev: No such file or directory
/usr/bin/ld: cannot find -ludev: No such file or directory
/usr/bin/ld: cannot find -ludev: No such file or directory
collect2: error: ld returned 1 exit status
gmake[2]: *** [examples/CMakeFiles/example_getCalibParamsFile.dir/build.make:151: /home/khughes1/new_unitree_cam/src/UnitreecameraSDK/bins/example_getCalibParamsFile] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:216: examples/CMakeFiles/example_getCalibParamsFile.dir/all] Error 2
/usr/bin/ld: cannot find -ludev: No such file or directory
/usr/bin/ld: cannot find -ludev: No such file or directory
/usr/bin/ld: cannot find -ludev: No such file or directory
collect2: error: ld returned 1 exit status
gmake[2]: *** [examples/CMakeFiles/example_getimagetrans.dir/build.make:151: /home/khughes1/new_unitree_cam/src/UnitreecameraSDK/bins/example_getimagetrans] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:268: examples/CMakeFiles/example_getimagetrans.dir/all] Error 2
collect2: error: ld returned 1 exit status
gmake[2]: *** [examples/CMakeFiles/example_putImagetrans.dir/build.make:151: /home/khughes1/new_unitree_cam/src/UnitreecameraSDK/bins/example_putImagetrans] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:242: examples/CMakeFiles/example_putImagetrans.dir/all] Error 2
collect2: error: ld returned 1 exit status
gmake[2]: *** [examples/CMakeFiles/example_getRectFrame.dir/build.make:151: /home/khughes1/new_unitree_cam/src/UnitreecameraSDK/bins/example_getRectFrame] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:190: examples/CMakeFiles/example_getRectFrame.dir/all] Error 2
collect2: error: ld returned 1 exit status
collect2: error: ld returned 1 exit status
gmake[2]: *** [examples/CMakeFiles/example_getRawFrame.dir/build.make:151: /home/khughes1/new_unitree_cam/src/UnitreecameraSDK/bins/example_getRawFrame] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:138: examples/CMakeFiles/example_getRawFrame.dir/all] Error 2
gmake[2]: *** [examples/CMakeFiles/example_getDepthFrame.dir/build.make:151: /home/khughes1/new_unitree_cam/src/UnitreecameraSDK/bins/example_getDepthFrame] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:164: examples/CMakeFiles/example_getDepthFrame.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
---
Failed   <<< UnitreeCameraSDK [3.81s, exited with code 2]

Summary: 1 package finished [4.00s]
  1 package failed: UnitreeCameraSDK
  1 package had stderr output: UnitreeCameraSDK
  1 package not processed

```

This was happening because the UnitreeCameraSDK requires OpenCv version 4.1.1 whereas my system used version 4.5. To get around this, I compiled version 4.1.1 and used that to set LD_LIBRARY_PATH as shown in the image below. 

```
LD_LIBRARY_PATH={path to older version of opencv}/lib:$LD_LIBRARY_PATH ros2 launch unitree_camera ...
```

![IMG_4382](https://github.com/ngmor/unitree_camera/assets/53623710/92d90c1c-b8b4-44a0-85cf-7649a66bdad2)

**Comment by ngmor** [comment]
Looks like the problem for `-ludev` is solved by running `sudo apt install libudev-dev`. I'll add this to the README.

**Comment by ngmor** [comment]
As for the OpenCV 4.1.1 dependency, I added some environment hooks to `UnitreecameraSDK` and some CMake variables that allow you to point to the specific OpenCV directory where you've installed 4.1.1. This, as well as new launch files for the Xavier's bottom camera, are all included as of aaadca0.
